Sources/PenguinStructures/NumberOperations.swift

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extension BinaryInteger {
+  /// Returns a sequence of the positive integers that are co-prime with `self`.
+  ///
+  /// Definition: Two numbers are co-prime if their GCD is 1.
+  public var positiveCoprimes: PositiveCoprimes<Self> { PositiveCoprimes(self) }
+}
+
+/// A sequence of numbers that are co-prime with `n`, up to `n`.
+public struct PositiveCoprimes<Number: BinaryInteger>: Sequence {
+  /// The number to find co-primes relative to.
+  let n: Number
+
+  /// Constructs a `PositiveCoprimes` sequence of numbers co-prime relative to `n`.
+  internal init(_ n: Number) {
+    precondition(n > 0, "\(n) doees not have defined positive co-primes.")
```

Comment:
I don't think that's right: https://www.wolframalpha.com/input/?i=gcd%282%2C+-3%29

If you need to make it a precondition that n>0, that's fine.

Also, though, there's nothing wrong with an empty sequence!

Reply:
Good catch! I think the right answer from a mathematical perspective is to convert to positive, and go from there. (Coprimes are symmetric around 0, so I don't think we need to define the corresponding negative coprimes sequence as a type, but it might be good to define a `.lazy.map { $0 * -1 }` extension... WDYT?